peer/connector.go

```diff
@@ -18,52 +18,69 @@ import (
 	"github.com/bitmark-inc/bitmarkd/messagebus"
 	"github.com/bitmark-inc/bitmarkd/mode"
 	"github.com/bitmark-inc/bitmarkd/peer/upstream"
+	"github.com/bitmark-inc/bitmarkd/peer/voting"
 	"github.com/bitmark-inc/bitmarkd/util"
 	"github.com/bitmark-inc/logger"
 )
 
 // various timeouts
 const (
-	cycleInterval         = 15 * time.Second // pause to limit bandwidth
-	connectorTimeout      = 60 * time.Second // time out for connections
-	samplelingLimit       = 10               // number of cycles to be 1 block out of sync before resync
-	fetchBlocksPerCycle   = 200              // number of blocks to fetch in one set
-	forkProtection        = 60               // fail to fork if height difference is greater than this
-	minimumClients        = 3                // do not proceed unless this many clients are connected
-	maximumDynamicClients = 10               // total number of dynamic clients
-)
+	// pause to limit bandwidth
+	cycleInterval = 15 * time.Second
 
-// a state type for the thread
-type connectorState int
+	// time out for connections
+	connectorTimeout = 60 * time.Second
 
-// state of the connector process
-const (
-	cStateConnecting   connectorState = iota // register to nodes and make outgoing connections
-	cStateHighestBlock connectorState = iota // locate node(s) with highest block number
-	cStateForkDetect   connectorState = iota // read block hashes to check for possible fork
-	cStateFetchBlocks  connectorState = iota // fetch blocks from current or fork point
-	cStateRebuild      connectorState = iota // rebuild database from fork point (config setting to force total rebuild)
-	cStateSampling     connectorState = iota // signal resync complete and sample nodes to see if out of sync occurs
+	// number of cycles to be 1 block out of sync before resync
+	samplelingLimit = 10
+
+	// number of blocks to fetch in one set
+	fetchBlocksPerCycle = 200
+
+	// fail to fork if height difference is greater than this
+	forkProtection = 60
+
+	// do not proceed unless this many clients are connected
+	minimumClients = 5
+
+	// total number of dynamic clients
+	maximumDynamicClients = 10
+
+	// client should exist at least 1 response with in this number
+	activePastSec = 60
 )
 
+type ConnectorIntf interface {
```

Comment:
I think should use `Connector` and `ConnectorImpl` for more consistent.

Reply:
Already have similar discussion

Reply:
But you are inconsistent. We need a consistence when coding, sometimes you use `AIntf` to define interface and `A` for the implementation. Sometimes you use `A` as interface and `AImpl` as implementation. Is that ok?

Reply:
@hxw How do you think?

Reply:
That's why I intend not to change others, because this module will be different as others, as you already demonstrate.

Reply:
Why we have inconsistence in the same code base? I don't have any reason to self-explain it. You are modifying the module and it breaks the code base convention, it's better if you follow the convention.

Reply:
I already say, for test, so I use interface.

Reply:
I means the inconsistent naming. I agree with you about creating interface for testing.